dev/com.ibm.ws.microprofile.config.1.3_fat/fat/src/com/ibm/ws/microprofile/config13/test/HotAddMPConfig.java

```diff
@@ -0,0 +1,149 @@
+ */
+@RunWith(FATRunner.class)
+public class HotAddMPConfig extends FATServletClient {
+
+    public static final String APP_NAME = "hotAddMPConfigApp";
+
+    @Server("HotAddMPConfig")
+    //@TestServlet(servlet = MapEnvVarServlet.class, contextRoot = APP_NAME)
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests r = RepeatConfigActions.repeatConfig13("HotAddMPConfig");
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create a WebArchive that will have the file name 'app1.war' once it's written to a file
+        // Include the 'app1.web' package and all of it's java classes and sub-packages
+        // Automatically includes resources under 'test-applications/APP_NAME/resources/' folder
+        // Exports the resulting application to the ${server.config.dir}/apps/ directory
+        ShrinkHelper.defaultApp(server, APP_NAME, "com.ibm.ws.microprofile.config.hotadd.*");
+        server.copyFileToLibertyServerRoot("HotAddMPConfig/noMPConfig/HotAddMPConfig.xml");
+        server.copyFileToLibertyServerRoot("HotAddMPConfig/withApp/HotAddApp.xml");
+        server.startServerAndValidate(true, true, false); //don't validate because the app won't have started properly
+        //server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer("CWNEN0047W.*HotAddMPConfigServlet");
+    }
+
+    /**
+     * Copy a server config file to the server root and wait for notification that the server config has been updated
+     *
+     * @param filename
+     * @throws Exception
+     */
+    private static void copyConfigFileToLibertyServerRoot(String filename, String appName) throws Exception {
+        server.setMarkToEndOfLog();
+        server.copyFileToLibertyServerRoot(filename);
+
+        if (appName == null) {
+            server.waitForConfigUpdateInLogUsingMark(null, false);
+        } else {
+            server.waitForConfigUpdateInLogUsingMark(Collections.singleton(appName), false);
+        }
+
+        Thread.sleep(ConfigConstants.DEFAULT_DYNAMIC_REFRESH_INTERVAL * 2); // We need this pause so that the MP config change is picked up through the polling mechanism
+    }
+
+//    @Before
+//    public void resetConfigFile() throws Exception {
+//        copyConfigFileToLibertyServerRoot("HotAddMPConfig/noMPConfig/HotAddMPConfig.xml", null);
+//        copyConfigFileToLibertyServerRoot("HotAddMPConfig/withApp/HotAddApp.xml", null);
+//    }
+
+    @Test
+    public void testHotAddMPConfig() throws Exception {
+        copyConfigFileToLibertyServerRoot("HotAddMPConfig/withMPConfig/HotAddMPConfig.xml", null);
+        //copyConfigFileToLibertyServerRoot("HotAddMPConfig/withApp/HotAddApp.xml", APP_NAME);
+
+        test(server, "/hotAddMPConfigApp/HotAddMPConfigServlet?testMethod=configTest");
+    }
+
+    private void test(LibertyServer server, String testUri) throws Exception {
```

Comment:
Probably makes more sense to use `FATServletClient.runTest()` rather than writing your own test method.

Reply:
done